server/schedule/plugin_interface.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schedule
+
+import (
+	"path/filepath"
+	"plugin"
+	"sync"
+
+	"github.com/pingcap/log"
+	"go.uber.org/zap"
+)
+
+// PluginMapLock is a read/write mutex for pluginMap
+var PluginMapLock = sync.RWMutex{}
+
+// PluginMap is a map for storing plugin(plugin-path -> plugin)
+var PluginMap = make(map[string]*plugin.Plugin)
+
+// GetFunction gets func by funcName from plugin(.so)
+func GetFunction(path string, funcName string) (plugin.Symbol, error) {
```

Comment:
Why not use the OOP style to wrap `GetFunction` `PluginMap` and `PluginMapLock`?